src/charm.py

```diff
@@ -1619,10 +1800,11 @@ def _pull_licenses(container: Container) -> None:
 
         for license_name in licenses:
             try:
-                license_file = container.pull(path=Config.get_license_path(license_name))
-                f = open(f"LICENSE_{license_name}", "x")
-                f.write(str(license_file.read()))
-                f.close()
+                # Lazy copy, only if the file wasn't already pulled.
+                filename = Path(f"LICENSE_{license_name}")
+                if not filename.is_file():
```

Comment:
Do we still need the `try/expect` if we have this check?

Reply:
Oh indeed should be fine removing it!
I'll try and confirm :)

Reply:
Fixed : 8973031f